test.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+set -e
+
+cd ~
+
+# Enter a virtualenv we control!
+python -m virtualenv .
+source bin/activate
+pip install --upgrade pip
+
+# install ros and some dependencies
+sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
+sudo apt-key adv --keyserver hkp://pgp.mit.edu:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
```

Comment:
No, its the public key for ros. This is essentially adding the PPA for ROS to our apt-get and then installing, see the buzzmobile install script for a similar example.